### PR TITLE
TicketBAI QR code and identifier generation without depending on the API

### PR DESCRIPTION
**Pull request by abdallahbeshi**
## Description

TicketBAI QR code and identifier generation without depending on the API.

This makes it possible to generate the QR code and TBAI identifier without depending on the API.
P.S it is still required to call the SendInvoices action to report the invoices, but this makes it possible not to always depend on the API call in order to generate the required invoice data.
(can be helpful if the API is down, or if it returns some errors).

## Type of change
- [ ] Bug fix.
- [ ] Feature.
- [x] Consolidation.

## Checklist
- [ ] Is breaking change.
- [ ] Documentation updated.
- [ ] Tests included (please specify cases).

### CODE REVIEW

**Review by marioporcic — DISMISSED** [review]

File: src/Basque/Mews.Fiscalizations.Basque/Mews.Fiscalizations.Basque.csproj [hunk 1]
```diff
@@ -10,7 +10,7 @@
     <RepositoryUrl>https://github.com/MewsSystems/fiscalizations</RepositoryUrl>
     <Icon>https://raw.githubusercontent.com/msigut/eet/master/receipt.png</Icon>
     <GeneratePackageOnBuild>true</GeneratePackageOnBuild>
-    <PackageVersion>2.0.3</PackageVersion>
+    <PackageVersion>2.0.4</PackageVersion>
```
Comment (marektresnak) [inline src/Basque/Mews.Fiscalizations.Basque/Mews.Fiscalizations.Basque.csproj:13]: Following the [semantic versioning](https://semver.org/), shouldn't we bump the minor version on changes like this?

File: src/Basque/Mews.Fiscalizations.Basque/Model/TbaiInvoiceData.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    {
+        public TbaiInvoiceData(string tbaiIdentifier, string qrCodeUri)
+        {
+            TbaiIdentifier = tbaiIdentifier;
+            QrCodeUri = qrCodeUri;
+        }
+
+        public string TbaiIdentifier { get; }
+
+        public string QrCodeUri { get; }
```
Comment (marektresnak) [inline src/Basque/Mews.Fiscalizations.Basque/Model/TbaiInvoiceData.cs:13]: QrCodeUri holds the actual data? Is it still an URI then?

File: src/Basque/Mews.Fiscalizations.Basque/TicketBaiClient.cs [hunk 3]
```diff
@@ -34,31 +33,64 @@ public TicketBaiClient(X509Certificate2 certificate, Region region, Environment
...
+
+        /// <summary>
+        /// Generates the QR code and TBAI Identifier which must be displayed on the invoice without calling the API.
+        /// To report the invoice to the gov authorities, SendInvoiceAsync must be used.
+        /// </summary>
+        /// <param name="request">Invoice request which will be mapped to Dto.TicketBai.</param>
+        /// <returns></returns>
+        public TbaiInvoiceData GetTbaiInvoiceData(SendInvoiceRequest request)
+        {
+            var signedRequest = GetSignedInvoiceDocument(request);
```
Comment (marektresnak) [inline src/Basque/Mews.Fiscalizations.Basque/TicketBaiClient.cs:62]: Nit: This is called twice in `SendInvoiceAsync()`.

**Review by marektresnak — DISMISSED** [review]
I don't know enough to verify the actual implementation 😄

File: src/Basque/Mews.Fiscalizations.Basque/Model/TbaiInvoiceData.cs (hunk 2 above)
Comment (abdallahbeshi) [reply src/Basque/Mews.Fiscalizations.Basque/Model/TbaiInvoiceData.cs:13]: Its actually a URI, it doesn't contain the QR data.

**Review by marektresnak — APPROVED** [review]

**Pull request merged**